clippy_lints/src/operators/arithmetic.rs

```diff
@@ -5,13 +5,21 @@
 
 use super::ARITHMETIC;
 use clippy_utils::{consts::constant_simple, diagnostics::span_lint};
+use rustc_ast as ast;
 use rustc_data_structures::fx::FxHashSet;
 use rustc_hir as hir;
 use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty::Ty;
 use rustc_session::impl_lint_pass;
-use rustc_span::source_map::Span;
+use rustc_span::source_map::{Span, Spanned};
 
-const HARD_CODED_ALLOWED: &[&str] = &["std::num::Saturating", "std::string::String", "std::num::Wrapping"];
+const HARD_CODED_ALLOWED: &[&str] = &[
+    "f32",
+    "f64",
```

Comment:
This is surprising to me, I would have expected this lint to be a superset of `float_arithmetic` and `integer_arithmetic`. It should be called out by name that floats aren't checked at least in the lint description

Reply:
An oversight. Floats don't "overflow" as regular integers, therefore, they aren't fit for this lint.

`float_arithmetic` even states that it is intended for kernels or embedded systems -> https://rust-lang.github.io/rust-clippy/master/index.html#float_arithmetic

Reply:
Yeah it makes sense, just didn't fit my initial expectation from the lint name

Reply:
Perhaps something like `side_effects_arithmetic`?